velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -97,14 +114,30 @@ ParquetRowReader::ParquetRowReader(
     : reader_(std::move(reader)), pool_(pool) {
   auto& selector = *options.getSelector();
   auto& filter = selector.getProjection();
+
   rowType_ = selector.buildSelectedReordered();
+  duckdbRowType_.reserve(rowType_->size());
 
   std::vector<::duckdb::column_t> columnIds;
   columnIds.reserve(rowType_->size());
-  duckdbRowType_.reserve(rowType_->size());
-  for (auto& node : filter) {
-    columnIds.push_back(node.column);
-    duckdbRowType_.push_back(reader_->return_types[node.column]);
+  for (uint64_t i = 0; i < filter.size(); i++) {
+    uint64_t columnId = filter[i].column;
+    VELOX_CHECK(
+        columnId < reader_->names.size(),
+        "Unexpected column Id {} for name: {}",
+        columnId,
+        filter[i].name);
+    columnIds.push_back(columnId);
+
+    // DuckDB ParquetReader return_types is actually the type for all columns in
```

Comment:
Comments should be full sentences, e.g. start with a capital letter and end with a period.

```
// DuckDB ParquetReader::return_types contains all columns present in the file.
```